pkg/cluster/cluster_config.go

```diff
@@ -287,3 +288,32 @@ func IsDefaultAuthMethod(ctx context.Context, cli client.Client) (bool, error) {
 	// we only create userGroups for "IntegratedOAuth" or "" and leave other or new supported type value in the future
 	return authenticationobj.Spec.Type == configv1.AuthenticationTypeIntegratedOAuth || authenticationobj.Spec.Type == "", nil
 }
+
+func GetNodeArchitectures(ctx context.Context, client client.Client) (map[string]struct{}, error) {
+	nodeList := &corev1.NodeList{}
+	if err := client.List(ctx, nodeList); err != nil {
+		return nil, fmt.Errorf("failed to list nodes: %w", err)
+	}
+
+	// Create a map to track unique architectures
+	nodeArchitectures := make(map[string]struct{})
+
+	for _, node := range nodeList.Items {
+		if arch, exists := node.Labels[labels.NodeArch]; exists {
+			if node.Status.Conditions != nil {
+				// Only count nodes that are Ready
+				for _, condition := range node.Status.Conditions {
+					if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionTrue {
+						nodeArchitectures[arch] = struct{}{}
+					}
+				}
+			}
+		}
+	}
+
+	if len(nodeArchitectures) < 1 {
+		return nil, errors.New("no valid architectures found")
+	}
+
+	return nodeArchitectures, nil
```

Comment:
probably here we want to return just the keys, not the entire map

Reply:
I initially opted for a map because it automatically handles any duplicate node arches and since the values of the map are empty structs with no fields, they take up zero bytes in memory. I'm happy to change it if you feel it's necessary but I think it keeping a `map[string]struct{}` will lead to less overall code with comparable memory usage to a `[]string` that will almost always have duplicate entries (since most clusters are single arch).

Reply:
I meant that the function signature should return a slice, while using a map internally, as you properly did.

I don't hold any strong opinion, so just for the discussion, my rationale is:
- the returned map doesn’t convey additional meaning since its values are empty, making it functionally equivalent to a slice.
-  if I understand correctly, the result is only iterated by key in `HasCompatibleArchitecture`, so returning a slice would simplify the API without losing efficiency.

Reply:
Sorry that's my mistake, I misunderstood your original comment but I see what you're saying now. I pushed the changes to fix it, thanks!